code/ui/Hud.cs

```diff
@@ -38,9 +45,9 @@ private void OnPlayerDied()
         }
 
         [Event("tttreborn.player.spawned")]
-        private void OnPlayerSpawned()
+        private void OnPlayerSpawned(TTTPlayer player)
         {
-            if (playerInfo != null || weaponSelection != null)
+            if (player != Local.Client.Pawn || playerInfo != null || weaponSelection != null)
```

Comment:
`player != Local.Client.Pawn`  seems to be an odd check here?

Reply:
Event `tttreborn.player.spawned` is called for every player. that's why :)